frontend/packages/dev-console/src/components/import/import-validation-utils.ts

```diff
@@ -64,18 +64,18 @@ export const detectGitType = (url: string): string => {
   return GitTypes.unsure;
 };
 
-export const detectGitRepoName = (url: string): string | undefined => {
-  if (!gitUrlRegex.test(url)) {
-    return undefined;
+export const createComponentName = (nameString: string): string => {
+  if (nameRegex.test(nameString)) {
+    return nameString;
   }
-
-  return _.kebabCase(url.split('/').pop());
+  const kebabCaseStr = _.kebabCase(nameString);
+  return nameString.match(/^\d/) ? `ocp-${kebabCaseStr}` : kebabCaseStr;
```

Comment:
```suggestion
  return nameString.match(/^\d/) || kebabCaseStr.match(/^\d/) ? `ocp-${kebabCaseStr}` : kebabCaseStr;
```
Technically, a github reposistory name can start with a hypen in the begining.
![image](https://user-images.githubusercontent.com/9964343/123945115-6dd94780-d9bb-11eb-8881-f7ef9ecf7975.png)

So if you call the createComponentName('-2dotnetcore') it returns back `2-dotnetcore` which is invalid name, so we need to consider the `kebabCaseStr` string as well, so that the output becomes `ocp-2-donetcore`